pkg/tcpip/stack/transport_demuxer.go

```diff
@@ -291,7 +291,7 @@ func (d *transportDemuxer) deliverPacket(r *Route, protocol tcpip.TransportProto
 	// Fail if we didn't find at least one matching transport endpoint.
 	if len(destEps) == 0 {
 		// UDP packet could not be delivered to an unknown destination port.
-		if protocol == header.UDPProtocolNumber {
+		if first == false && protocol == header.UDPProtocolNumber {
```

Comment:
Instead of `first == false`, how about just `!first`?

Reply:
Done, thanks!